### PR TITLE
Fixed bug in ConsumerStreamIterator.attachStream()

### PR DESCRIPTION
**Pull request by larryaasen**
Fixed bug in ConsumerStreamIterator.attachStream() where '==' was being used instead of '=', so the _subscription was not reset to null on rebalance. Also, renamed async.dart to consumer_streamiterator.dart which was more appropriate.

### CODE REVIEW

**Comment by larryaasen** [comment]
I don't understand why my branch passes on Travis, but not when it is run from here. Travis builds the same branch for both GitHub accounts, both mine passes on this one has one test failure.

Screen Shot 2020-04-25 at 9 10 51 AM

You can see the passing of commit ece25a2 which is my latest commit on this branch.

**Comment by larryaasen** [comment]
The consumer and producer seem very stable now, and I have been running them for about 24 hours. I have two processes: one to push out a message every 60 seconds, and the other to consume it.

The tool is setup now to run Kafka 0.10.2.1 dated April 26, 2017 using Docker image ches/kafka:latest. The next step would be to upgrade the tool to use a recent version of Docker. I attempted that but it requires significant image changes, because ches/kafka does not have a recent version of Kafka.

**Comment by larryaasen** [comment]
@pulyaevskiy  This latest update is using the latest version of Kafka 2.5.0, and passes all 48 tests. Please check this out.

File: lib/src/consumer_streamiterator.dart [hunk 1]
```diff
@@ -111,7 +106,7 @@ class ConsumerStreamIterator<K, V>
     Object records;
     if (_subscription != null) {
       _subscription.cancel();
-      _subscription == null;
```
Comment (pulyaevskiy) [inline lib/src/consumer_streamiterator.dart:114]: Ouch, good catch! :)

**Review by pulyaevskiy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**